pkg/core/sync.go

```diff
@@ -0,0 +1,258 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/cloudquery/cloudquery/internal/logging"
+	"github.com/cloudquery/cloudquery/pkg/core/database"
+	"github.com/cloudquery/cloudquery/pkg/core/state"
+	"github.com/cloudquery/cloudquery/pkg/plugin"
+	"github.com/cloudquery/cloudquery/pkg/plugin/registry"
+	sdkdb "github.com/cloudquery/cq-provider-sdk/database"
+	"github.com/cloudquery/cq-provider-sdk/migration"
+	"github.com/cloudquery/cq-provider-sdk/provider/diag"
+	"github.com/cloudquery/cq-provider-sdk/provider/execution"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+	"github.com/rs/zerolog/log"
+	"github.com/thoas/go-funk"
+)
+
+type SyncState int
+
+const (
+	Installed SyncState = iota + 1
+	Upgraded
+	Downgraded
+	NoChange
+)
+
+func (s SyncState) String() string {
+	switch s {
+	case Installed:
+		return "Installed"
+	case Upgraded:
+		return "Upgraded"
+	case Downgraded:
+		return "Downgraded"
+	case NoChange:
+		return "NoChange"
+	}
+	return "Unknown"
+}
+
+type SyncResult struct {
+	State      SyncState
+	OldVersion string
+	NewVersion string
+}
+
+const dropTableSQL = "DROP TABLE IF EXISTS %s CASCADE"
```

Comment:
nit: why not use goqu?

Reply:
this is more clear than unknown goqu. This was moved here from the SDK which had the line exactly like this.